### DISCUSSION TITLE
Use of nextauth for okta sso in a website that is embedded under another website - Reading cookie in cross-site context will be blocked in future Chrome versions Cookies with the SameSite=None; Secure and not Partitioned attributes that operate in cross-site contexts are third-party cookies. In future Chrome versions, reading third-party cookies will be blocked.

### DISCUSSION THREAD

**Discussion by pravinhabbu4u** [nextauthjs/next-auth#10023] [Help]
We have built a website which uses next-auth to handle SSO logins using okta. This website is now embedded by another application. We have done a few changes to the next-auth options to allow reading of cookies with sameSite=none and partitioned=true. Along with this we are using content security policy to further secure the embed functionality. Google still shows warning about "Cookies with the SameSite=None; Secure and not Partitioned attributes that operate in cross-site contexts are third-party cookies. In future Chrome versions, reading third-party cookies will be blocked. This behavior protects user data from cross-site tracking."
__Host-next-auth.csrf-token
__Secure-next-auth.callback-url
__Secure-next-auth.session-token
__cf_bm
Has anyone came across similar scenario and what is the fix for it? 
next-auth options:
cookies: {
    sessionToken: {
      name: '__Secure-next-auth.session-token',
      options: {
        httpOnly: true,
        sameSite: 'none',
        path: '/',
        secure: true,
        partitioned: true
      },
    },
    callbackUrl: {
      name: '__Secure-next-auth.callback-url',
      options: {
        sameSite: 'none',
        path: '/',
        secure: true,
        partitioned: true
      },
    },
    csrfToken: {
      name: '__Host-next-auth.csrf-token',
      options: {
        httpOnly: true,
        sameSite: 'none',
        path: '/',
        secure: true,
        partitioned: true
      },
    },
    state: {
      name: '__Secure-next-auth.state',
      options: {
        httpOnly: true,
        sameSite: 'none',
        path: '/',
        secure: true,
        partitioned: true
      },
    },
    pkceCodeVerifier: {
      name:'next-auth.pkce.code_verifier',
      options: {
        httpOnly: true,
        sameSite: 'none',
        path: '/',
        partitioned: true,
        secure: true,
      },
    }
  },

**Comment by brooksbecton** [comment]
Hey Pravin!

I just ran into this and have some solutions, but it depends on which version of next-auth you are using. 

If you are on the v5 beta, using the latest package version should have this fix. It came in here: [#9254](https://github.com/nextauthjs/next-auth/issues/9254)

If you are on a v4, then you'll need to make the change yourself. The change in the v5/main branch updates the `cookie` package from 0.5.0->0.60. 

You can do this yourself by adding this snippet to your package.json

```
  "overrides": {
    "next-auth": {
      "cookie": "0.6.0"
    }
  }
```

This update allows the "partitioned" param to go through and set in the cookie in next-auth v4. 
For any else's context, this will be required if you site is using third party cookies (Like and embed) if you want to use Google's CHIPS approach.
